### PR TITLE
7853 registration table

### PR DESCRIPTION
**Pull request by jordiwes**
*Issue #:* /bcgov/entity#7853

*Description of changes:*
- ui table with some enhancements, but not all
- some tests, but will improve upon tests as well

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts [hunk 1]
```diff
@@ -34,8 +34,35 @@ export interface FinancingStatementIF {
 // Draft interface. TODO: add change, amendment statement draft definitions when available.
 export interface DraftIF {
     type: string, // One of enum DraftTypes.
+    documentId?: string,
```
Comment (doug-lovett) [inline ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts]: DraftIF represents an API draft, which may be for a financing statement, a change statement, or an amendment statement. FinancingStatementIF holds all the API information for a draft financing statement. There is some duplication here, and in the RegistrationIF definition (which is not in the API). When an amendment draft is implemented it will be added to DraftIF as AmendmentStatementIF.

Comment (jordiwes) [reply ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts]: I see what you're saying here, but the documentId is not coming back under a financingStatement key 
financingStatement : { documentId: "abc" }
so I guess some translation would need to happen in the ppr-api helper...

File: ppr-ui/src/assets/styles/overrides.scss [hunk 2]
```diff
@@ -278,48 +285,121 @@ tbody {
...
+  font-size: 9px !important;
+}
+
+.registration-date .v-picker--date {
+  box-shadow: 0px 3px 1px -2px rgb(0 0 0 / 20%),
+    0px 2px 2px 0px rgb(0 0 0 / 14%), 0px 1px 5px 0px rgb(0 0 0 / 12%);
+}
+
+/*
+.registration-table th:last-child,
```
Comment (kialj876) [inline ppr-ui/src/assets/styles/overrides.scss]: saving for later or should this be deleted?

File: ppr-ui/src/components/registration/RegistrationTable.vue [hunk 3]
```diff
@@ -0,0 +1,676 @@
+<template>
```
Comment (kialj876) [inline ppr-ui/src/components/registration/RegistrationTable.vue]: How do you feel about moving this into the 'tables' folder?

File: ppr-ui/src/components/registration/RegistrationTable.vue [hunk 4]
```diff
@@ -0,0 +1,676 @@
...
+        <v-col cols="6">
+          <v-date-picker
+            color="#1A5A96"
+            :max="submittedEndDateTmp"
+            v-model="submittedStartDateTmp"
+          />
+        </v-col>
+        <v-col cols="6">
+          <v-date-picker
+            color="#1A5A96"
```
Comment (kialj876) [inline ppr-ui/src/components/registration/RegistrationTable.vue]: if there isn't a theme for this colour you should create one and use that for these instead I think

File: ppr-ui/src/components/registration/RegistrationTable.vue [hunk 5]
```diff
@@ -0,0 +1,676 @@
...
+    }
+  }
+})
+</script>
+
+<style lang="scss" module>
+@import '@/assets/styles/theme.scss';
+.reg-table {
+  max-height: 550px;
+  /*tbody > tr > td:nth-last-child(1),
```
Comment (kialj876) [inline ppr-ui/src/components/registration/RegistrationTable.vue]: Please remove or comment if it will be used later

File: ppr-ui/src/components/registration/RegistrationTable.vue [hunk 6]
```diff
@@ -0,0 +1,676 @@
...
+  position: relative;
+  top: -100px;
+  float: right;
+  height: 0px;
+}
+.text-input-style {
+  background-color: white !important;
+  border: 1px solid var(--outline);
+  height: 45px;
+  /* padding: 0 0 5px 8px; */
```
Comment (kialj876) [inline ppr-ui/src/components/registration/RegistrationTable.vue]: remove or comment pls

File: ppr-ui/src/composables/useRegistration.ts [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+        if (originalData[i].expireDays) {
+          if (originalData[i].expireDays === localState.daysToExpiry) {
+            newTableData[i].hide = true
+          }
+        }
+      }
+    }
+
+    if (localState.submittedStartDate) {
+      const startDate = new Date(localState.submittedStartDate)
```
Comment (kialj876) [inline ppr-ui/src/composables/useRegistration.ts]: you can probably get away with just comparing the string values of both dates instead of creating the date object each time (this might get slow for a large result set). What do you think?

File: ppr-ui/src/composables/useRegistration.ts [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+          const originalStartDate = new Date(originalData[i].createDateTime)
+          if (startDate > originalStartDate) {
+            newTableData[i].hide = true
+          }
+        }
+      }
+    }
+
+    if (localState.submittedEndDate) {
+      const endDate = new Date(localState.submittedEndDate)
```
Comment (kialj876) [inline ppr-ui/src/composables/useRegistration.ts]: same question as above

File: ppr-ui/src/resources/statusTypes.ts [hunk 9]
```diff
@@ -0,0 +1,10 @@
+export const StatusTypes: Array<any> = [
```
Comment (kialj876) [inline ppr-ui/src/resources/statusTypes.ts:3]: please define this array and add enums for 'text' / 'value' or put in a FUTURE comment to add them later

File: ppr-ui/src/utils/ppr-api-helper.ts [hunk 10]
```diff
@@ -227,66 +251,163 @@ export async function updateDraft (draft: DraftIF): Promise<DraftIF> {
...
+}
+
+// Get registration history
+export async function registrationHistory (): Promise<[RegistrationIF]> {
+  const url = sessionStorage.getItem('PPR_API_URL')
+  const config = { baseURL: url, headers: { Accept: 'application/json' } }
+  return axios
+    .get('financing-statements/registrations', config)
+    .then(response => {
+      console.log(response)
```
Comment (kialj876) [inline ppr-ui/src/utils/ppr-api-helper.ts]: please remove log

File: ppr-ui/src/utils/ppr-api-helper.ts [hunk 11]
```diff
@@ -227,66 +251,163 @@ export async function updateDraft (draft: DraftIF): Promise<DraftIF> {
...
+}
+
+// Get registration history
+export async function draftHistory (): Promise<[DraftIF]> {
+  const url = sessionStorage.getItem('PPR_API_URL')
+  const config = { baseURL: url, headers: { Accept: 'application/json' } }
+  return axios
+    .get('drafts', config)
+    .then(response => {
+      console.log(response)
```
Comment (kialj876) [inline ppr-ui/src/utils/ppr-api-helper.ts]: please remove log

File: ppr-ui/src/utils/ppr-api-helper.ts [hunk 12]
```diff
@@ -28,13 +29,20 @@ import { SearchHistoryResponseIF } from '@/interfaces/ppr-api-interfaces/search-
  * 500 INTERNAL_SERVER_ERROR
  */
 
-export const successfulPPRResponses = [StatusCodes.OK, StatusCodes.CREATED, StatusCodes.ACCEPTED]
+export const successfulPPRResponses = [
```
Comment (kialj876) [inline ppr-ui/src/utils/ppr-api-helper.ts:33]: why all the style changes in this file :( ? If it's passing the linter and you're not changing the logic then I think it's a lot nicer to leave this out of the diff. If you want to change things like this put it in a separate PR please

File: ppr-ui/src/composables/useRegistration.ts [hunk 13]
```diff
@@ -0,0 +1,285 @@
...
+    () => localState.registeredBy,
+    () => {
+      filterResults(localState.originalData)
+    }
+  )
+
+  watch(
+    () => localState.status,
+    () => {
+      console.log(localState.status)
```
Comment (kialj876) [inline ppr-ui/src/composables/useRegistration.ts]: please remove log

**Review by kialj876 — COMMENTED** [review]
Hey Steph, great work here! I have a lot of comments, but they are all small things

File: ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts (hunk 1 above)
Comment (jordiwes) [reply ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts]: I've reworked this now

File: ppr-ui/src/components/registration/RegistrationTable.vue (hunk 3 above)
Comment (jordiwes) [reply ppr-ui/src/components/registration/RegistrationTable.vue]: I feel fine with it, so I moved it

File: ppr-ui/src/assets/styles/overrides.scss (hunk 2 above)
Comment (jordiwes) [reply ppr-ui/src/assets/styles/overrides.scss]: yeah, I removed it

File: ppr-ui/src/components/registration/RegistrationTable.vue (hunk 4 above)
Comment (jordiwes) [reply ppr-ui/src/components/registration/RegistrationTable.vue]: changed

File: ppr-ui/src/components/registration/RegistrationTable.vue (hunk 5 above)
Comment (jordiwes) [reply ppr-ui/src/components/registration/RegistrationTable.vue]: all of my attempts at getting that last column sticky. removed.

File: ppr-ui/src/components/registration/RegistrationTable.vue (hunk 6 above)
Comment (jordiwes) [reply ppr-ui/src/components/registration/RegistrationTable.vue]: done

File: ppr-ui/src/utils/ppr-api-helper.ts (hunk 10 above)
Comment (jordiwes) [reply ppr-ui/src/utils/ppr-api-helper.ts]: done

File: ppr-ui/src/utils/ppr-api-helper.ts (hunk 11 above)
Comment (jordiwes) [reply ppr-ui/src/utils/ppr-api-helper.ts]: done

File: ppr-ui/src/utils/ppr-api-helper.ts (hunk 12 above)
Comment (jordiwes) [reply ppr-ui/src/utils/ppr-api-helper.ts:33]: I just probably ran the linter on it................................

File: ppr-ui/src/composables/useRegistration.ts (hunk 13 above)
Comment (jordiwes) [reply ppr-ui/src/composables/useRegistration.ts]: done

File: ppr-ui/src/resources/statusTypes.ts (hunk 9 above)
Comment (jordiwes) [reply ppr-ui/src/resources/statusTypes.ts:3]: done

File: ppr-ui/src/components/registration/RegistrationTable.vue (hunk 3 above)
Comment (kialj876) [reply ppr-ui/src/components/registration/RegistrationTable.vue]: thanks!

File: ppr-ui/src/utils/ppr-api-helper.ts (hunk 12 above)
Comment (kialj876) [reply ppr-ui/src/utils/ppr-api-helper.ts:33]: darn haha

**Review by kialj876 — APPROVED** [review]
Thanks for making those changes! Still have a question from before, but it is not a blocker

File: ppr-ui/src/composables/useRegistration.ts (hunk 7 above)
Comment (jordiwes) [reply ppr-ui/src/composables/useRegistration.ts]: I made the change

File: ppr-ui/src/composables/useRegistration.ts (hunk 8 above)
Comment (jordiwes) [reply ppr-ui/src/composables/useRegistration.ts]: made the change

**Review by doug-lovett — APPROVED** [review]
Looks great.
